url-shortener/src/main/java/com/bobocode/shortener/service/DefaultURLShortenService.java

```diff
@@ -0,0 +1,47 @@
+package com.bobocode.shortener.service;
+
+import com.bobocode.shortener.dto.URLPayload;
+import com.bobocode.shortener.entity.ShortenedUrl;
+import com.bobocode.shortener.repository.ShortenedUrlRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.concurrent.ThreadLocalRandom;
+
+@Service
+@RequiredArgsConstructor
+public class DefaultURLShortenService implements URLShortenService {
+
+    private static final String URL_TEMPLATE = "boboshort.";
+    private final ShortenedUrlRepository shortenedUrlRepository;
+
+    @Transactional
+    @Override
+    public String shortenUrl(final URLPayload payload) {
+        Objects.requireNonNull(payload);
+        final int randomValue = ThreadLocalRandom.current().nextInt(0, Integer.MAX_VALUE);
+        var url = URL_TEMPLATE + randomValue;
```

Comment:
@vromantsev this strategy is not good, because some of your url may be
`http://vlad.com/api/short/urls/boboshort.3` and while the other `http://vlad.com/api/short/urls/boboshort.1283234033`